apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1376,23 +1374,113 @@ new_onchain_tx(channel_withdraw_tx, #{acct := ToId,
                                to_id      => aeser_id:create(account, ToId)
                               }),
     lager:debug("withdraw_tx Opts = ~p", [Opts1]),
-    {ok, WithdrawTx} = aesc_withdraw_tx:new(Opts1),
+    {ok, WithdrawTx} = new_onchain_tx_(aesc_withdraw_tx, Opts1, CurrHeight),
     {ok, WithdrawTx, Updates};
-
 new_onchain_tx(channel_create_tx, Opts,
                #data{opts = #{initiator := Initiator,
                               responder := Responder},
-                     state=State}) ->
+                     state=State}, CurrHeight) ->
     StateHash = aesc_offchain_state:hash(State),
-    Opts1 = maps:merge(Opts, #{initiator => Initiator,
-                               responder => Responder}),
+    Opts1 = Opts#{ state_hash    => StateHash
+                 , initiator_id  => aeser_id:create(account, Initiator)
+                 , responder_id  => aeser_id:create(account, Responder)
+                 },
     lager:debug("create_tx Opts = ~p", [Opts1]),
-    Opts2 = Opts1#{ state_hash    => StateHash
-                  , initiator_id  => aeser_id:create(account, Initiator)
-                  , responder_id  => aeser_id:create(account, Responder)
-                  },
-    {ok, CreateTx} = aesc_create_tx:new(Opts2),
-    {ok, CreateTx, []}. % no updates
+    {ok, CreateTx} = new_onchain_tx_(aesc_create_tx, Opts1, CurrHeight),
+    {ok, CreateTx, []}; % no updates
+new_onchain_tx(channel_settle_tx, Opts,
+               #data{ opts  = #{initiator := I,
```

Comment:
We should decide which formatting to use in case of multi-line maps and records. 6 lines above the alternative is used.

Reply:
Tracked as [Decide on multi-line formatting](https://www.pivotaltracker.com/story/show/167484806)